### PR TITLE
Chore/redo error handling

### PR DESCRIPTION
**Pull request by tarkah**
## Proposed Changes
  - Better concrete error types out of `ajour_core`
  - Adding human readable context to errors at application level and only displaying that to end users via gui
  - Better logging of errors at application level, logging both human readable context and underlying error

## Checklist

- [x] Tested on Windows
- [x] Tested on MacOS
- [x] Tested on Linux
- [x] Added an entry to `CHANGELOG.md` if knowledge of this change could be valuable to users

### CODE REVIEW

File: crates/core/src/lib.rs [hunk 1]
```diff
@@ -12,7 +12,3 @@ pub mod repository;
 #[cfg(feature = "gui")]
 pub mod theme;
 pub mod utility;
-
-use crate::error::ClientError;
-
-pub type Result<T> = std::result::Result<T, ClientError>;
```
Comment (casperstorm) [inline crates/core/src/lib.rs:18]: Nice we got rid of this.

Comment (tarkah) [reply crates/core/src/lib.rs:18]: yup! We have a few different error types now

**Comment by casperstorm** [comment]
No more comments here, this is awesome.

**Review by casperstorm — APPROVED** [review]

**Comment by casperstorm** [comment]
You can merge once you feel this is ready, since its a big one @tarkah.

**Comment by tarkah** [comment]
Sounds good! I'll back merge dev since we've added some more stuff and get it merged later

**Pull request merged**